src/firebase-db.ts

```diff
@@ -271,6 +282,121 @@ export const watchAllAnswers = (callback: (wrappedAnswer: WrappedDBAnswer[]) =>
   });
 };
 
+const getActivityLevelFeedbackDocsQuery = () => {
+  if (!portalData) {
+    throw new Error("Must set portal data first");
+  }
+  // Only logged-in students will have feedback.
+  if (portalData.userType !== "learner" || portalData.type === "anonymous") return;
+
+  let query: firebase.firestore.Query = app.firestore().collection(teacherFeedbackPath("activity"));
+
+  if (portalData.type === "authenticated") {
+    query = query
+      .where("platformId", "==", portalData.platformId)
+      .where("resourceLinkId", "==", portalData.resourceLinkId)
+      .where("contextId", "==", portalData.contextId)
+      .where("platformStudentId", "==", portalData.platformUserId.toString());
+  }
+
+  return query;
+};
+
+const getQuestionLevelFeedbackDocsQuery = (answerId: string) => {
+  if (!portalData) {
+    throw new Error("Must set portal data first");
+  }
+  // Only logged-in students will have feedback.
+  if (portalData.userType !== "learner" || portalData.type === "anonymous") return;
+
+  let query: firebase.firestore.Query = app.firestore().collection(teacherFeedbackPath("question"));
+
+  if (portalData.type === "authenticated") {
+    query = query
+      .where("answerId", "==", answerId)
```

Comment:
Did you need to leave in the answerId?

Reply:
No. Thanks for catching that. It's gone now.